cedar-policy-core/src/ast/name.rs

```diff
@@ -414,7 +416,67 @@ impl FromStr for Name {
     }
 }
 
+lazy_static::lazy_static! {
+    // PANIC SAFETY: this is a valid regex
+    static ref ID_REGEX: Regex = Regex::new(r"^[_a-zA-Z][_a-zA-Z0-9]*(?:::[_a-zA-Z][_a-zA-Z0-9]*)*$").unwrap();
+    // All Cedar's reserved keywords
+    static ref RESERVED: HashSet<&'static str> =
```

Comment:
This is somewhat different from the list of reserved keywords for Cedar `id` (as documented [here](https://github.com/cedar-policy/cedar/blob/6d01368f8d341b5da97b9fb0c6aa18b5a76b306a/cedar-policy-core/src/ast/id.rs#L29)).  It would be nice to have a unit test that checks the validity (or not) of `Name`s containing all of the keywords on either of these lists.